src/Main.elm

```diff
@@ -45,24 +114,24 @@ view : Model -> Html Msg
 view model =
     div [ class "calculator" ]
         [ div [] [ Html.node "style" [] [ text Css.css ] ]
-        , div [ class "display" ] [ text "display" ]
+        , div [ class "display" ] [ text model.display ]
         , div [ class "buttons" ]
             [ button [ class "operator" ] [ text "+" ]
             , button [ class "operator" ] [ text "-" ]
             , button [ class "operator" ] [ text "/" ]
             , button [ class "operator" ] [ text "*" ]
-            , button [ class "number" ] [ text "1" ]
-            , button [ class "number" ] [ text "2" ]
-            , button [ class "number" ] [ text "3" ]
-            , button [ class "number" ] [ text "4" ]
-            , button [ class "number" ] [ text "5" ]
-            , button [ class "number" ] [ text "6" ]
-            , button [ class "number" ] [ text "7" ]
-            , button [ class "number" ] [ text "8" ]
-            , button [ class "number" ] [ text "9" ]
-            , button [ class "number" ] [ text "0" ]
+            , button [ class "number", onClick (UpdateDisplay calNumOne) ] [ text calNumOne ]
```

Comment:
Does this code compile? You have a function called `updateDisplay` defined above. But here you have `UpdateDisplay`. I doubt this code compiles.

Reply:
oops, I see the constructor now!

Reply:
very good... carry on :)

Reply:
I would emphasize @jbrechtel 's point then that the `UpdateDisplay` constructor then should wrap a type of your own definition, not just a String.

Reply:
e.g.
```
type Button = One | Two | ...
```
then you can have a function that maps values of type Button to String if you like
```
displayButton : Button -> String
displayButton = _ 
```